tests/app/questionnaire/test_path_finder.py

```diff
@@ -535,3 +536,253 @@ def test_new_remove_answer_and_block_if_routing_backwards(list_store):
+        ),
+        (
+            "No",
+            None,
+            "test_new_routing_and_skipping_section_dependencies",
+            "skip-confirmation-section",
+            ["security"],
+        ),
+        (
+            "No",
+            None,
+            "test_new_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block", "age", "reason-no-confirmation"],
+        ),
+        (
+            "Yes",
+            "Yes",
+            "test_new_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block"],
+        ),
+        (
+            "Yes",
+            "No",
+            "test_new_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block", "age"],
+        ),
+        (  # The skipping age question must of changed from Yes to No after answering the confirmation question,
+            # this means the confirmation answer will not be on the path, so the primary person section will asked why there was no confirmation
+            "No",
+            "Yes",
+            "test_new_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block", "age", "reason-no-confirmation"],
+        ),
+        (
+            "Yes",
+            None,
+            "test_routing_and_skipping_section_dependencies",
+            "skip-confirmation-section",
+            ["security", "skip-confirmation"],
+        ),
+        (
+            "Yes",
+            None,
+            "test_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block", "reason-no-confirmation"],
+        ),
+        (
+            "No",
+            None,
+            "test_routing_and_skipping_section_dependencies",
+            "skip-confirmation-section",
+            ["security"],
+        ),
+        (
+            "No",
+            None,
+            "test_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block", "age", "reason-no-confirmation"],
+        ),
+        (
+            "Yes",
+            "Yes",
+            "test_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block"],
+        ),
+        (
+            "Yes",
+            "No",
+            "test_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block", "age"],
+        ),
+        (
+            "No",
+            "Yes",
+            "test_routing_and_skipping_section_dependencies",
+            "primary-person",
+            ["name-block", "age", "reason-no-confirmation"],
+        ),
+    ),
+)
+def test_routing_path_block_ids_dependent_on_when_rules(
+    list_store,
+    skip_age_answer,
+    skip_confirmation_answer,
+    schema,
+    section_id,
+    expected_route,
+):
+    # Given
+    schema = load_schema_from_name(schema)
```

Comment:
This test is pretty hard to digest IMO as it is specific to the schema set up. Maybe some comments would help. I think when coming back to this in the future, will won't be very obvious on why this is doing the things it does.

Reply:
Possibly (and I know I've not been doing it because the test params are fairly straightforward), you can add id's to `@pytest.mark.parametrize(..., ids=['skip age question results in skip confirmation question'], ...)`. I try and avoid where possible though because the list of id's should correspond in order to the test params which can get disordered easily. Solution would be to define a data structure which contains the params and the ids together then passed to the parametrize decorator but then there's a bit more stuff to grok when reviewing and maintaining.